packages/docusaurus/src/server/i18n.ts

```diff
@@ -15,7 +15,10 @@ function getDefaultLocaleLabel(locale: string) {
   // Intl.DisplayNames is ES2021 - Node14+
   // https://v8.dev/features/intl-displaynames
   if (typeof Intl.DisplayNames !== 'undefined') {
-    return new Intl.DisplayNames([locale], {type: 'language'}).of(locale);
+    const languageName = new Intl.DisplayNames(locale, {type: 'language'}).of(
+      locale,
+    );
+    return languageName.charAt(0).toLocaleUpperCase() + languageName.slice(1);
```

Comment:
```suggestion
    return (
      languageName.charAt(0).toLocaleUpperCase(locale) +
      languageName.substring(1)
    );
```

Two things:

1. The locale should certainly be the locale that the language is converted to, not the runner's host locale
2. I prefer `substring` to `slice` but feel free to disagree

Reply:
Thanks!

Reply:
> The locale should certainly be the locale that the language is converted to, not the runner's host locale

Yes that's what I meant, my comment was probably not very explicit :D